manifests/init.pp

```diff
@@ -174,11 +174,12 @@
 
     # These regexes match absolute paths or paths that begin with an existing
     # puppet configuration variable, like $vardir
-    validate_re($classfile,'^(\$(?!/)|/).+')
-    validate_re($confdir,'^(\$(?!/)|/).+')
-    validate_re($environmentpath,'^(\$(?!/)|/).+')
-    validate_re($logdir,'^(\$(?!/)|/).+')
-    validate_re($rundir,'^(\$(?!/)|/).+')
+
+    assert_type(Pattern['^(\$(?!/)|/).+'], $classfile)
+    assert_type(Pattern['^(\$(?!/)|/).+'], $confdir)
```

Comment:
$confdir, $logdir, and $rundir are typed to be Stdlib::Absolutepath.  So puppet configuration variables will not be allowed and these regexes are unnecessary.  Does this mean the types above are incorrect?

Reply:
Removed the asserts that are now obsolete